src/EditorFeatures/Core/Implementation/ChangeSignature/AbstractChangeSignatureCommandHandler.cs

```diff
@@ -87,13 +87,24 @@ private static void ExecuteCommand(ITextView textView, ITextBuffer subjectBuffer
                 return;
             }
 
-            // Reorder parameters is not cancellable.
-            var reorderParametersService = document.GetLanguageService<AbstractChangeSignatureService>();
-            var result = reorderParametersService.ChangeSignature(
-                document,
-                caretPoint.Value.Position,
-                (errorMessage, severity) => workspace.Services.GetService<INotificationService>().SendNotification(errorMessage, severity: severity),
-                CancellationToken.None);
+            ChangeSignatureResult result = null;
+            var waitResult = _waitIndicator.Wait(
+                FeaturesResources.Change_signature,
+                allowCancel: true,
```

Comment:
Change-sig is rather interesting in that the expensive computation is up front.  So that's the part i put under a dialog.